modules/firestore/src/main/scala/zio/gcp/firestore/package.scala

```diff
@@ -1,68 +1,53 @@
 package zio.gcp
 
-import com.google.cloud.firestore._
-import zio.RIO
+import com.google.cloud.firestore.{ Firestore => _, _ }
 
-package object firestore extends FirestoreDB.Service[FirestoreDB] {
+import zio.{ Has, RIO }
 
-  def batch: RIO[FirestoreDB, WriteBatch] = RIO.accessM(_.firestore.batch)
+package object firestore {
 
-  def collection(collectionPath: CollectionPath): RIO[FirestoreDB, CollectionReference] =
-    RIO.accessM(_.firestore.collection(collectionPath))
+  type Firestore = Has[Firestore.Service]
 
-  def collectionGroup(collectionPath: CollectionPath): RIO[FirestoreDB, Query] =
-    RIO.accessM(_.firestore.collectionGroup(collectionPath))
+  def allCollections: RIO[Firestore, List[CollectionReference]] =
+    RIO.accessM(_.get.allCollections)
 
-  def commit(batch: WriteBatch): RIO[FirestoreDB, List[WriteResult]] =
-    RIO.accessM(_.firestore.commit(batch))
+  def allDocuments(collectionPath: CollectionPath): RIO[Firestore, List[QueryDocumentSnapshot]] =
+    RIO.accessM(_.get.allDocuments(collectionPath))
+
+  def batch: RIO[Firestore, WriteBatch] =
+    RIO.accessM(_.get.batch)
+
+  def collection(collectionPath: CollectionPath): RIO[Firestore, CollectionReference] =
+    RIO.accessM(_.get.collection(collectionPath))
+
+  def collectionGroup(collectionPath: CollectionPath): RIO[Firestore, Query] =
+    RIO.accessM(_.get.collectionGroup(collectionPath))
+
+  def commit(batch: WriteBatch): RIO[Firestore, List[WriteResult]] =
+    RIO.accessM(_.get.commit(batch))
 
   def createDocument[A](
-    collectionReference: CollectionReference,
-    documentPath: DocumentPath,
-    document: A
-  ): RIO[FirestoreDB, WriteResult] =
-    RIO.accessM(_.firestore.createDocument(collectionReference, documentPath, document))
-
-  def delete(
-    collectionPath: CollectionPath,
-    documentPath: DocumentPath
-  ): RIO[FirestoreDB, WriteResult] =
-    RIO.accessM(_.firestore.delete(collectionPath, documentPath))
-
-  def document(
-    collectionReference: CollectionReference,
-    documentPath: DocumentPath
-  ): RIO[FirestoreDB, DocumentReference] =
-    RIO.accessM(_.firestore.document(collectionReference, documentPath))
-
-  def getDocumentSnapshot(
-    collectionPath: CollectionPath,
-    documentPath: DocumentPath
-  ): RIO[FirestoreDB, DocumentSnapshot] =
-    RIO.accessM(_.firestore.getDocumentSnapshot(collectionPath, documentPath))
-
-  def getCollections: RIO[FirestoreDB, List[CollectionReference]] =
-    RIO.accessM(_.firestore.getCollections)
-
-  def getAllDocuments(
-    collectionPath: CollectionPath
-  ): RIO[FirestoreDB, List[QueryDocumentSnapshot]] =
-    RIO.accessM(
-      _.firestore
-        .getAllDocuments(collectionPath)
-    )
-
-  def set[A](
-    collectionPath: CollectionPath,
+    collectionRef: CollectionReference,
     documentPath: DocumentPath,
-    document: A
-  ): RIO[FirestoreDB, WriteResult] =
-    RIO.accessM(_.firestore.set(collectionPath, documentPath, document))
+    data: A
```

Comment:
Revert back to `document` to be consistent with the terminology of the underlying library